src/bin/load_test.rs

```diff
@@ -305,118 +466,118 @@ async fn broadcast_random_documents_load_test(
+
+                // Wait for the rate limiter to allow further processing
+                tokio::select! {
+                   _ = rate_limiter.until_ready() => {},
+                   _ = cancel_task.cancelled() => return,
+                };
+
+                // Broadcast the document
+                tracing::trace!(
+                    "broadcasting document {}",
+                    random_document.id().to_string(Encoding::Base58),
+                );
 
-            let result = random_document
-                .put_to_platform(
-                    &sdk,
-                    document_type.as_ref().clone(),
-                    document_state_transition_entropy,
-                    identity_public_key,
-                    &signer,
-                    Some(PutSettings {
-                        request_settings: settings,
-                        identity_nonce_stale_time_s: None,
-                    }),
-                )
-                .await;
-
-            pending.fetch_sub(1, Ordering::SeqCst);
-
-            match result {
-                Ok(_) => {
-                    oks.fetch_add(1, Ordering::SeqCst);
-
-                    tracing::trace!(
-                        "document {} successfully broadcast",
-                        random_document.id().to_string(Encoding::Base58),
-                    );
-                }
-                Err(error) => {
-                    tracing::error!(
-                        ?error,
-                        "failed to broadcast document {}: {}",
-                        random_document.id().to_string(Encoding::Base58),
-                        error
-                    );
+                pending.fetch_add(1, Ordering::SeqCst);
 
-                    errs.fetch_add(1, Ordering::SeqCst);
-                }
-            };
+                let elapsed_secs = start_time.elapsed().as_secs();
 
-            drop(permit);
-        });
+                if start_time.elapsed().as_secs() % 10 == 0
+                    && elapsed_secs != last_report.load(Ordering::SeqCst)
+                {
+                    tracing::info!(
+                        "{} secs passed: {} pending, {} successful, {} failed",
+                        elapsed_secs,
+                        pending.load(Ordering::SeqCst),
+                        oks.load(Ordering::SeqCst),
+                        errs.load(Ordering::SeqCst),
+                    );
+                    last_report.swap(elapsed_secs, Ordering::SeqCst);
+                }
 
-        tasks.push(task)
+                let result = random_document
+                    .put_to_platform(
+                        &sdk,
+                        document_type_to_use.as_ref().clone(),
+                        document_state_transition_entropy,
+                        identity_public_key,
+                        signer.as_ref(),
+                        Some(PutSettings {
+                            request_settings: settings,
```

Comment:
This does not build for me due to too many arguments. Please specify required version of Sdk in Cargo.toml.